AsyncDisplayKit/ASControlNode.mm

```diff
@@ -273,37 +259,33 @@ - (void)addTarget:(id)target action:(SEL)action forControlEvents:(ASControlNodeE
       });
     }
   }
+  
+  // Create new target action pair
+  ASControlTargetAction *targetAction = [[ASControlTargetAction alloc] init];
+  targetAction.action = action;
+  targetAction.target = target;
 
   // Enumerate the events in the mask, adding the target-action pair for each control event included in controlEventMask
   _ASEnumerateControlEventsIncludedInMaskWithBlock(controlEventMask, ^
     (ASControlNodeEvent controlEvent)
     {
       // Do we already have an event table for this control event?
       id<NSCopying> eventKey = _ASControlNodeEventKeyForControlEvent(controlEvent);
-      NSMapTable *eventDispatchTable = _controlEventDispatchTable[eventKey];
-      // Create it if necessary.
-      if (!eventDispatchTable)
-      {
-        // Create the dispatch table for this event.
-        eventDispatchTable = [NSMapTable weakToStrongObjectsMapTable];
-        if (eventKey) {
-          [_controlEventDispatchTable setObject:eventDispatchTable forKey:eventKey];
-        }
+      NSMutableArray *eventTargetActionArray = _controlEventDispatchTable[eventKey];
+      
+      if (!eventTargetActionArray) {
+        eventTargetActionArray = [[NSMutableArray alloc] init];
       }
-
-      // Have we seen this target before for this event?
-      NSMutableSet *targetActions = [eventDispatchTable objectForKey:target];
-      if (!targetActions)
-      {
-        // Nope. Create an action set for it.
-        targetActions = [[NSMutableSet alloc] initWithCapacity:kASControlNodeActionDispatchTableInitialCapacity]; // enough to handle common types without re-hashing the dictionary when adding entries.
-        [eventDispatchTable setObject:targetActions forKey:target];
+      
+      // Remove any prior target-action pair for this event, as UIKit does.
+      [eventTargetActionArray removeObject:targetAction];
```

Comment:
@tamaskarai it would be better to do removeObjectIdenticalTo:, I think.  Not a critical issue, but this code is much more performance-sensitive than you might think — because all ASImageNodes, ASTextNodes, ASButtonNodes are ASControlNodes and they are invoked heavily during scrolling.